tiledb/sm/array/array.cc

```diff
@@ -683,6 +683,10 @@ Status Array::metadata(Metadata** metadata) {
   return Status::Ok();
 }
 
+uint64_t Array::num_threads() const {
```

Comment:
It appears that this is only used in the subarray.cc. Instead of exposing this through `Array::num_threads()` and `StorageManager::num_threads()`, how about we construct `SubarrayPartitioner` instances with the number of threads and pass that directly into the routines that need it.

Reply:
We may need it elsewhere actually. This will probably be used in all `parallel_for` loops over ranges, as currently `parallel_for` constructs a vector of status for all ranges. We have been benchmarking with billions and trillions of such ranges, so that creates memory problems. Checking the concurrency we split the ranges in batches, so that `parallel_for` operates always on `num_threads` number of items.

Reply:
That seems like it will be messy to pass this all around the code, it really seems like a utility constant that could be globally accessible, but I don't see another way since it is tied to the individual storage manager configurations.

Reply:
Just a note here that this is not a constant. I am trying to get the number of TBB threads set by dynamically the user here, which only the `StorageManager` knows through the `Config` it stores.

Reply:
And current the `StorageManager` is stored in the `Array`, which is stored in the `Subarray`.